src/icons/patterns/utils.ts

```diff
@@ -0,0 +1,61 @@
+function addExts(name: string, exts: string[]): string[] {
+  return exts.map((ext) => name + '.' + ext);
+}
+
+type Method = 'CONF' | 'ECMA' | 'BOTH';
+
+const ext: Record<Method, string[]> = {
+  CONF: ['json', 'jsonc', 'json5', 'yaml', 'yml', 'toml'],
+  ECMA: ['js', 'mjs', 'cjs', 'ts', 'mts', 'cts'],
+  get BOTH() {
+    return [...this.CONF, ...this.ECMA];
+  },
+};
+
+type patternFunc = (key: string) => string[];
+
+const to: Record<Method | 'AsRC', patternFunc> = {
+  ECMA: (name) => addExts(name, ext.ECMA),
+  CONF: (name) => addExts(name, ext.CONF),
+  BOTH: (name) => addExts(name, ext.BOTH),
+  AsRC: (name) => [name, ...addExts(name, ext.BOTH)],
+};
+
+export type PatternType =
+  | 'ECMA'
+  | 'CONF'
+  | 'BOTH'
+  | 'DotNameConfig'
+  | 'NameDotConfig'
+  | 'DotConfigName'
+  | 'DotNameRC'
+  | 'DotConfigNameRC'
+  | 'Cosmiconfig'
+  | 'CommonlyUsed';
```

Comment:
Offering 10 patterns, but only use `Cosmiconfig` for now, other patterns for discussions